data/src/main/java/com/linkedin/data/schema/validation/ValidateDataAgainstSchema.java

```diff
@@ -241,6 +241,7 @@ else if (element instanceof MutableDataElement)
            validatorElement = new SimpleDataElement(fixed, element.getName(), schema, element.getParent());
         }
         _context._dataElement = validatorElement;
+        _context._dataSchema = schema;
```

Comment:
This schema is already available as context::dataElement::schema. Why can't you use that?
(see line 227 above)